core/utils.lua

```diff
@@ -388,73 +388,93 @@ function serialize_string(s)
 	G:save_settings()
 	G.FUNCS.update_atlas(args.cycle_config.type)
 end
 
 -- Set the atlases of all cards of the correct type to be the new palette
 G.FUNCS.update_atlas = function(type)
+    local name = G.SETTINGS.selected_texture[type]
+	if not SMODS.AltTextures[type][name] then return end
 	local atlas_keys = {}
-	if type == "Suits" then
+	if type == "Suit" then
 		atlas_keys = {"cards_1", "ui_1"}
-		G.C["SO_1"].Clubs = G.SETTINGS.selected_colours[type].new_colours[1] or G.C["SO_1"].Clubs
-		G.C["SO_1"].Spades = G.SETTINGS.selected_colours[type].new_colours[2] or G.C["SO_1"].Spades
-		G.C["SO_1"].Diamonds = G.SETTINGS.selected_colours[type].new_colours[3] or G.C["SO_1"].Diamonds
-		G.C["SO_1"].Hearts = G.SETTINGS.selected_colours[type].new_colours[4] or G.C["SO_1"].Hearts
-		G.C.SUITS = G.C.SO_1
-			
+		for suit, _ in pairs(G.C["SO_1"]) do
+			local colour = (SMODS.AltTextures.Suit[name].suits and SMODS.AltTextures.Suit[name].suits[suit] or SMODS.AltTextures.Suit["Default"].suits[suit] or nil)
+			G.C["SO_1"][suit] =  (colour and HEX(colour) or G.C["SO_1"][suit])
+			G.C.SUITS[suit] = G.C["SO_1"][suit]
+		end		
+    elseif type == "Seal" then
+        atlas_keys = {"centers"}
+    elseif type == "Tag" then
+        atlas_keys = {"tags"}
+    elseif type == "Blind" then
+        if G.ASSET_ATLAS[type][name] then
+            G.ANIMATION_ATLAS["blind_chips"].image = G.ASSET_ATLAS[type][name].image
+        else
+            sendDebugMessage("Can't find type "..type.." with name "..name)
+            print(tprint(G.ASSET_ATLAS[type]))    
+        end
+        return
 	else
-		for _,v in pairs(G.P_CENTER_POOLS[type]) do
-			atlas_keys[v.atlas or type] = v.atlas or type
+		for _, card in pairs(G.P_CENTER_POOLS[type]) do
+			atlas_keys[card.atlas or type] = card.atlas or type
 		end
+        if type == "Spectral" then
+            atlas_keys["soul"] = "soul"
+        end
 	end
-	for _,v in pairs(atlas_keys) do
-		if G.ASSET_ATLAS[v][G.SETTINGS.selected_colours[type].name] then
-			G.ASSET_ATLAS[v].image = G.ASSET_ATLAS[v][G.SETTINGS.selected_colours[type].name].image
+	for _, atlas_key in pairs(atlas_keys) do
+		if G.ASSET_ATLAS[atlas_key][name] then
+			G.ASSET_ATLAS[atlas_key].image = G.ASSET_ATLAS[atlas_key][name].image
 		end
 	end
 end
 
-G.FUNCS.card_colours = function(e)
-    G.SETTINGS.paused = true
-    G.FUNCS.overlay_menu{
-      definition = G.UIDEF.card_colours(),
-    }
-  end
-
-G.UIDEF.card_colours = function()
-    local nodeRet = {}
-    for _,k in ipairs(SMODS.Palettes.Types) do
-		local v = SMODS.Palettes[k]
-        if #v.names > 1 then
-            nodeRet[#nodeRet+1] = create_option_cycle({w = 4,scale = 0.8, label = k.." colours" ,options = v.names, opt_callback = "update_recolor", current_option = G.SETTINGS.selected_colours[k].order, type=k})
-        end
-    end
-    local t = create_UIBox_generic_options({back_func = 'options', contents = nodeRet})
-    return t
-end
-
 G.FUNCS.recolour_image = function(x,y,r,g,b,a)
```

Comment:
Not sure what this is doing now

Reply:
this is the function that recolours base textures